torchtext/csrc/sentencepiece.cpp

```diff
@@ -5,51 +5,69 @@
 // #include <sentencepiece_processor.h>
 
 namespace torchtext {
-/*
-namespace {
+
 struct SentencePiece : torch::CustomClassHolder {
- private:
+private:
   sentencepiece::SentencePieceProcessor processor_;
- public:
-  // public because we use it during registration below.
-  std::string content_;
-  explicit SentencePiece(std::string content) : content_(content) {
-    auto status = processor_.LoadFromSerializedProto(content);
+public:
+  std::string model_proto_;
+  SentencePiece() {}
+  /*
+  SentencePiece(const std::string& model) {
+    processor_.LoadFromSerializedProto(model);
+  }
+  */
+  void Load(const std::string& path) {
+    const auto status = processor_.Load(path);
     if (!status.ok()) {
-        std::stringstream ss;
-        ss << "Failed to load SentencePiece model. Error: ";
-        ss << status.ToString();
-        throw std::runtime_error(ss.str());
+      // TODO: Use proper logging
+      std::cerr << status.ToString() << std::endl;
+      // TODO: Either fail here or propagate the error to Python.
```

Comment:
pybind catches and translates cpp exceptions: https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html